### PR TITLE
Granular CRN for network datasources and resources

### PR DESCRIPTION
**Pull request by Alexander-Kita**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Relates OR Closes #0000

Output from acceptance testing:

pi_network (d):

```
=== RUN   TestAccIBMPINetworkDataSource_basic
--- PASS: TestAccIBMPINetworkDataSource_basic (12.23s)
PASS
```

pi_networks (d):

```
=== RUN   TestAccIBMPINetworksDataSource_basic
--- PASS: TestAccIBMPINetworksDataSource_basic (11.58s)
PASS
```

pi_public_network (d):

```
=== RUN   TestAccIBMPIPublicNetworkDataSource_basic
--- PASS: TestAccIBMPIPublicNetworkDataSource_basic (9.65s)
PASS
```

pi_network (r):

```
=== RUN   TestAccIBMPINetworkbasic
--- PASS: TestAccIBMPINetworkbasic (57.95s)
PASS
```

```
=== RUN   TestAccIBMPINetworkGatewaybasic
--- PASS: TestAccIBMPINetworkGatewaybasic (77.08s)
PASS
```

```
=== RUN   TestAccIBMPINetworkDHCPbasic
--- PASS: TestAccIBMPINetworkDHCPbasic (65.87s)
PASS
```

pi_network_port_attach (r):

```
=== RUN   TestAccIBMPINetworkPortAttachVlanbasic
--- PASS: TestAccIBMPINetworkPortAttachVlanbasic (63.98s)
PASS
```

### CODE REVIEW

File: ibm/service/power/resource_ibm_pi_network.go [hunk 1]
```diff
@@ -159,7 +171,10 @@ func resourceIBMPINetworkCreate(ctx context.Context, d *schema.ResourceData, met
 			body.DNSServers = networkdns
 		}
 	}
-
+	if tags, ok := d.GetOk(Arg_UserTags); ok {
+		userTags := flex.ExpandStringList(tags.([]interface{}))
+		body.UserTags = userTags
+	}
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_network.go:177]: Could check the Len()>0 before expanding?

File: ibm/service/power/resource_ibm_pi_network_port_attach.go [hunk 2]
```diff
@@ -101,7 +108,10 @@ func resourceIBMPINetworkPortAttachCreate(ctx context.Context, d *schema.Resourc
 		ipaddress := v.(string)
 		nwportBody.IPAddress = ipaddress
 	}
-
+	if tags, ok := d.GetOk(Arg_UserTags); ok {
+		userTags := flex.ExpandStringList(tags.([]interface{}))
+		nwportBody.UserTags = userTags
+	}
```
Comment (michaelkad) [inline ibm/service/power/resource_ibm_pi_network_port_attach.go:114]: Same as above

**Review by michaelkad — CHANGES_REQUESTED** [review]
Left few comments

**Review by ismirlia — APPROVED** [review]

**Review by michaelkad — APPROVED** [review]

**Review by michaelkad — APPROVED** [review]

**Comment by Alexander-Kita** [comment]
Opened in main repo

**Pull request closed without merging**